### ISSUE TITLE
LSP reports error when psalm from CLI doesn't

### ISSUE THREAD

**Issue by zmitic** [vimeo/psalm#4819]
For WeakMap stub like this:

```php
/**
 * @template TKey
 * @template TValue
 *
 * @implements \ArrayAccess<TKey, TValue>
 * @implements \IteratorAggregate<TKey, TValue>
 */
final class WeakMap implements \ArrayAccess, \Countable, \IteratorAggregate
{
    public function getIterator(): iterable;
}
```

and usage similar to

```php
class MyClass
{
    /**
     * @var WeakMap<AggregatedEntityInterface, bool>
     */
    private WeakMap $watchedEntities;

    final public function reset(): void
    {
        /** @var WeakMap<AggregatedEntityInterface, bool> $wm */
        $wm = new WeakMap();
        $this->watchedEntities = $wm;
    }
}
```

doesn't report any problems when psalm is run from command line. But LSP shows this:

![image](https://user-images.githubusercontent.com/1964158/101633093-15def180-3a27-11eb-8693-c414a910978b.png)

Removing redundant ``@var``, reports this problem:

```
$this->watchedEntities expects 

'WeakMap<App\Entity\Model\Aggregates\AggregatedEntityInterface, bool>',  parent type `WeakMap<mixed, mixed>` provided see https://psalm.dev/196)
        $this->watchedEntities = $wm;

```

![image](https://user-images.githubusercontent.com/1964158/101633470-9140a300-3a27-11eb-958b-9145de86c45f.png)

---

It is just annoyance, not a big deal but ArrayCollection (from Doctrine) doesn't report this. If something is wrong with my stub, is there a better way?

Thanks.

---

Reproducer doesn't work for PHP8: https://psalm.dev/r/eb410a7347

Is there some way to set stubs when creating one?

**Comment by muglug** [comment]
That message isn't coming from Psalm's LSP

**Comment by weirdan** [comment]
That's PHPStorm's own inspection, see https://www.jetbrains.com/help/phpstorm/php-redundant-var-tag.html

**Comment by zmitic** [comment]
I am so sorry guys, you are right. Too used to red color coming from LSP.

Is it possible for psalm to infer it so I don't get this error:

![image](https://user-images.githubusercontent.com/1964158/101648174-d8379400-3a39-11eb-936f-54ad7085b253.png)

**Comment by weirdan** [comment]
> Is it possible for psalm to infer it

Not at the moment, and unlikely anytime soon, but now that PHPStorm claims to support Psalm annotattions you probably could open a bug on jetbrains tracker about false positive redundant var - as it's not really redundant.
